dipy/denoise/patch2self.py

```diff
@@ -1,50 +1,73 @@
+import atexit
+import os
+import tempfile
 import time
 from warnings import warn
 
 import numpy as np
+from tqdm import tqdm
 
-import dipy.core.optimize as opt
 from dipy.testing.decorators import warning_for_keywords
 from dipy.utils.optpkg import optional_package
 
 sklearn, has_sklearn, _ = optional_package("sklearn")
 linear_model, _, _ = optional_package("sklearn.linear_model")
 
 
-def _vol_split(train, vol_idx):
-    """Split the 3D volumes into the train and test set.
+def count_sketch(matrixA_name, matrixA_dtype, matrixA_shape, sketch_rows, tmp_dir):
+    """Count Sketching algorithm to reduce the size of the matrix.
 
     Parameters
     ----------
-    train : ndarray
-        Array of all 3D patches flattened out to be 2D.
-
-    vol_idx: int
-        The volume number that needs to be held out for training.
+    matrixA_name : str
+        The name of the memmap file containing the matrix A.
+    matrixA_dtype : dtype
+        The dtype of the matrix A.
+    matrixA_shape : tuple
+        The shape of the matrix A.
+    s : int
+        The number of rows in the sketch matrix.
+    tmp_dir : str
+        The directory to save the temporary files.
+    sketch_rows : int
+        The number of rows in the sketch matrix.
```

Comment:
Please, respect the order of the parameters in the method signature. Check this in the rest of the methods that have been changed.